src/Databases/DatabaseReplicated.cpp

```diff
@@ -520,11 +518,14 @@ void DatabaseReplicated::startupTables(ThreadPool & thread_pool, LoadingStrictne
     chassert(!TSA_SUPPRESS_WARNING_FOR_READ(tables_metadata_digest));
     TSA_SUPPRESS_WARNING_FOR_WRITE(tables_metadata_digest) = digest;
 
-    if (is_probably_dropped)
-        return;
+    tryConnectToZooKeeperAndInitDatabase(mode);
+    if (!is_probably_dropped)
+    {
+        ddl_worker = std::make_unique<DatabaseReplicatedDDLWorker>(this, getContext());
+        ddl_worker->startup();
+    }
 
-    ddl_worker = std::make_unique<DatabaseReplicatedDDLWorker>(this, getContext());
-    ddl_worker->startup();
+    DatabaseAtomic::startupTables(thread_pool, mode);
```

Comment:
@tavplubix can you please take a look that it is safe? If not, then some atomic flag can be introduced instead, that will be checked in the `canExecuteReplicatedMetadataAlter`

Reply:
We should avoid executing the DDL queue while tables are not started up yet, so an atomic flag would be better

Reply:
OK, let's go this way - the patch had been updated.